plugin/builtin/patchtransformer/PatchTransformer.go

```diff
@@ -143,9 +143,12 @@ func (p *plugin) applySMPatch(resource, patch *resource.Resource) error {
 	if err != nil {
 		return err
 	}
-	return filtersutil.ApplyToJSON(patchstrategicmerge.Filter{
+	n := resource.GetName()
```

Comment:
PatchTarget doesn't only contain name but also namespace.

https://github.com/kubernetes-sigs/kustomize/blob/45893b258899b5adaada80a29e59770afcdf3143/api/types/patchtarget.go#L11-L15

Reply:
So should we save the old namespace as well, in addition to the name? Anything else I should save? I think in meeting you also mentioned saving the GVK

Reply:
GVK is probably not needed since the patch shouldn't have different GVK otherwise it won't match the target. The reason that we need name and namespace is name and namespace can be modified in the pipeline and the name/namespace may be the old one in the patch.

Reply:
updated to include namespace